sfge/src/test/java/com/salesforce/graph/ops/expander/ApexPathExpanderTest.java

```diff
@@ -622,4 +624,67 @@ public void testInlineResolvableTwoLevelMethodCall() {
         MatcherAssert.assertThat(results, hasSize(equalTo(1)));
         MatcherAssert.assertThat(results, TestRunnerListMatcher.hasValuesAnyOrder("hello"));
     }
+
+    @Disabled // TODO: Handle nested Maps and invocations of get() on the results of a get() on a nested Map.
+    @Test
+    public void testNullPointerExceptionInOptimization_moreComplex() {
```

Comment:
Where is the NPE coming from? Could you add a comment above the line of source code that throws the NPE?

Reply:
The test probably doesn't need "NullPointerException" in its name anymore since we know the root cause. Will rename this.